File: Miscellaneous/Golang/13. Roman to Integer.go
package main

import (
	"fmt"
	"strings"
)

func main() {

	str := "IX"
	result := romanToInt(str)
	fmt.Println(result)
}

func romanToInt(s string) int {
	str := strings.Split(s, "")
	res := 0

	for i := 0; i < len(str); i++ {
		str1 := charToInt(str[i])
		if i+1 < len(str) {
			str2 := charToInt(str[i+1])
			if str2 <= str1 {
				res += str1
			} else {
				res += str2
				res -= str1
				i++
			}
		} else {
			res += str1
		}
	}

	return res
}

func charToInt(s string) int {
	var num int

	switch s {
	case "I":
		num = 1
		break
	case "V":
		num = 5
		break
	case "X":
		num = 10
		break
	case "L":
		num = 50
		break
	case "C":
		num = 100
		break
	case "D":
		num = 500
		break
	case "M":
		num = 1000
		break
	}
	return num
}

File: Miscellaneous/Python/13. Roman to Integer.py
class Solution:
    def romanToInt(self, s):
        romans = {
        "I":1,
        "V":5,
        "X":10,
        "L":50,
        "C":100,
        "D":500,
        "M":1000,
        }

        res = 0
        for i in range(len(s)):
            str1 = romans[s[i]]
            if i+1 < len(s):
                str2 = romans[s[i+1]]
                if str1 >= str2:
                    res += str1
                else:
                    res += str2
                    res -= str1
                    i=i+1
            else:
                res += str1

        return res

ans = Solution()
print(ans.romanToInt("XIX"))